src/main/java/project/bookstore/config/ApplicationConfiguration.java

```diff
@@ -0,0 +1,46 @@
+package project.bookstore.config;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.config.annotation.authentication.configuration.AuthenticationConfiguration;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import project.bookstore.repository.UserRepository;
+
+@Configuration
+@RequiredArgsConstructor
+public class ApplicationConfiguration {
+    private final UserRepository userRepository;
+
+    @Bean
+    public UserDetailsService userDetailsService() {
```

Comment:
it is better to make your own implementation instead of a bean for UserDetailsService